src/main/java/sirius/web/security/ScopeInfo.java

```diff
@@ -208,124 +207,131 @@ public boolean is(@Nonnull Class<?> type) {
 
-        throw Exceptions.handle()
-                        .to(UserContext.LOG)
-                        .withSystemErrorMessage("Cannot make a helper of type %s", aClass.getName())
-                        .handle();
+        return makeHelperByConstructor(aClass);
     }
 
-    private Object makeHelperByName(String name) {
-        for (HelperFactory<?> factory : factories) {
-            if (Strings.areEqual(name, factory.getName())) {
-                return makeHelper(factory);
-            }
+    private Object makeHelperByConstructor(Class<?> aClass) {
+        // First try to instantiate with a constructor that takes ScopeInfo as argument....
+        try {
+            Constructor<?> constructor = aClass.getDeclaredConstructor(ScopeInfo.class);
+            return constructor.newInstance(this);
+        } catch (NoSuchMethodException | IllegalAccessException e) {
+            // There is either no constructor or it isn't accessible -> ignore
+            Exceptions.ignore(e);
+        } catch (InstantiationException | InvocationTargetException e) {
+            throw Exceptions.handle()
+                            .to(UserContext.LOG)
+                            .error(e)
+                            .withSystemErrorMessage("Cannot auto instantiate a helper of type %s - %s (%s)",
+                                                    aClass.getName())
+                            .handle();
         }
 
+        // Then try to instantiate with a no args constructor....
+        try {
+            Constructor<?> constructor = aClass.getDeclaredConstructor();
+            return constructor.newInstance();
+        } catch (NoSuchMethodException | IllegalAccessException e) {
+            // There is either no constructor or it isn't accessible -> ignore
+            Exceptions.ignore(e);
+        } catch (InstantiationException | InvocationTargetException e) {
+            throw Exceptions.handle()
+                            .to(UserContext.LOG)
+                            .error(e)
+                            .withSystemErrorMessage("Cannot auto instantiate a helper of type %s - %s (%s)",
+                                                    aClass.getName())
+                            .handle();
+        }
+
+        // There is no way to instantiate this helper - give up...
         throw Exceptions.handle()
                         .to(UserContext.LOG)
-                        .withSystemErrorMessage("Cannot make a helper named %s", name)
+                        .withSystemErrorMessage("Cannot make a helper of type %s", aClass.getName())
                         .handle();
     }
 
-    private Object makeHelper(HelperFactory<?> factory) {
-        return makeHelper(factory, 0);
-    }
-
-    private Object makeHelper(HelperFactory<?> factory, int circuitBreaker) {
-        circuitBreaker++;
+    private Object populateHelper(Class<?> type, Object helper, Map<Class<?>, Object> localContext) {
+        ctx.wire(helper);
```

Comment:
could we also call this `context` or `globalContext` during this refactoring?

Reply:
done